lsp/src/Language/LSP/Server/Processing.hs

```diff
@@ -404,11 +404,8 @@ handle' logger mAction m msg = do
     -- See https://microsoft.github.io/language-server-protocol/specifications/specification-current/#-notifications-and-requests.
     reportMissingHandler :: m ()
     reportMissingHandler =
-      let optional = isOptionalNotification m
+      let optional = isOptionalMethod (SomeMethod m)
       in logger <& MissingHandler optional m `WithSeverity` if optional then Warning else Error
-    isOptionalNotification (SMethod_CustomMethod p)
-      | "$/" `T.isPrefixOf` T.pack (symbolVal p) = True
-    isOptionalNotification _  = False
```

Comment:
I've built HLS with this fix, everything seems fine till now. But could you please tell me what's `symbolVal` wrong here, does `symbolVal` only returns `SetTrace`?

Reply:
A custom method has a `KnownSymbol` dictionary inside it that we can use to get the method string, since we can't know it statically. For the other methods, we know it statically. `someMethodToMethodString` handles all the cases. `SetTrace` used to be a custom method (i.e. we didn't list it explicitly), but now we do (because it's in the metamodel), so this code was wrong because it only worked for custom methods.